### DISCUSSION TITLE
Expected argument of type "int", "object" given at property path "groupId".

### DISCUSSION THREAD

**Discussion by frenci8** [symfony/symfony#42703] [Q&A]
Hi, I am stuck at a form error and I can not seem to understand what the problem is. I have the following code

```php
    $builder
        ->add('username', HiddenType::class, [
            'data' => $options['username']
        ])
        ->add('groupId', EntityType::class, [
            'class' => Groups::class,
            'choices' => $options['groups'],
            'choice_value' => function($object) {
                return $object ? (int) $object->getId() : 0;
            },
            'choice_label' => function($object) {
                return $object ? $object->getName() : '';
            }
        ])
    ;
```

where `$options['groups']`  is just an array of Groups entity objects and groupId in the entity accepts only int type values. The form is rendered fine, if I check with inspect element, select element is created and the options have the correct value and label, but when I click on submit for the form, I keep getting this error `Expected argument of type "int", "object" given at property path "groupId".`  , but why? I call the method getId() in the choice_value option and even cast it to int in case something wrong is happening, but I still get the error that I am sending an object. How do I fix it? I could not find anything in the docs about dynamic values. If I try an example with static values, I do not get the object error, it works fine.

This is what $options['groups'] looks like:

```php
        Array
        (
            [0] => App\Entity\Groups Object
                (
                    [id:App\Entity\Groups:private] => 4
                    [name:App\Entity\Groups:private] => group 4
                )
        
            [1] => App\Entity\Groups Object
                (
                    [id:App\Entity\Groups:private] => 5
                    [name:App\Entity\Groups:private] => group 5
                )
        
            [2] => App\Entity\Groups Object
                (
                    [id:App\Entity\Groups:private] => 6
                    [name:App\Entity\Groups:private] => group 6
                )
        
            [3] => App\Entity\Groups Object
                (
                    [id:App\Entity\Groups:private] => 7
                    [name:App\Entity\Groups:private] => group 7
                )
        
            [4] => App\Entity\Groups Object
                (
                    [id:App\Entity\Groups:private] => 11
                    [name:App\Entity\Groups:private] => group 1
                )
        
        )
```

Anyone has any ideas? 

Thank you

**Comment by Nek-** [comment]
Hello, I'm sorry but your message is a real pain to read. Can you please reformat with [markdown](https://guides.github.com/features/mastering-markdown/)?

It's also missing a lot of information. From what I read I imagine you have an entity `User` and an entity `Group` and the issue is located on the doctrine mapping because you didn't define `groupId` as a doctrine relationship. I can be obviously wrong because it's an assumption with barely any information but the error message.
